pkg/minishift/network/settings_windows.go

```diff
@@ -33,13 +35,62 @@ const (
 	networkingMessageName = "PROVISION_NETWORKING"
 )
 
-func ConfigureNetworking(machineName string, vmDriver string, networkSettings NetworkSettings) {
+func determineNameservers() []string {
+	posh := powershell.New()
+	defer posh.Close()
+
+	command := `Get-DnsClientServerAddress | Where-Object { $_.InterfaceAlias -like 'vEthernet*' } | ForEach-Object { $_.ServerAddresses }`
+	result, _ := posh.Execute(command)
+
+	nameservers := strings.Split(result, "\r\n")
+	ipv4ns := []string{}
+	for _, ns := range nameservers {
+		if net.ParseIP(ns).To4() != nil {
+			ipv4ns = append(ipv4ns, ns)
+		}
+	}
+
+	return ipv4ns
+}
+
+func ConfigureNetworking(machineName string, networkSettings NetworkSettings) {
 	// Instruct the user that this does not work for other Hypervisors on Windows
-	if vmDriver != "hyperv" {
+	if !minishiftConfig.IsHyperV() {
 		fmt.Println(configureIPAddressMessage, configureIPAddressFailure)
 		return
 	}
 
+	if networkSettings.Device == "" {
+		networkSettings.Device = "eth0"
+	}
+
+	if networkSettings.Gateway == "" {
+		fmt.Println("-- Determing default gateway ... ")
+		networkSettings.Gateway = determineDefaultGateway(networkSettings.IPAddress)
+	}
+
+	if networkSettings.DNS1 == "" && networkSettings.DNS2 == "" {
+		fmt.Printf("-- Determing nameservers to use ... ")
+		nameservers := determineNameservers()
+
+		if len(nameservers) == 0 {
+			fmt.Println("FAIL")
+			fmt.Println("   Consider to configure using '--network-nameserver'")
+		}
+
+		if len(nameservers) > 0 {
+			fmt.Println("OK")
+			networkSettings.DNS1 = nameservers[0]
+		}
+		if len(nameservers) > 1 {
+			networkSettings.DNS2 = nameservers[1]
+		}
+		if len(nameservers) > 2 {
+			fmt.Println("   WARN: found more than 2 nameservers")
```

Comment:
is having more than 2 name servers is a bad thing?

Reply:
It is, as when the host will return more, we are actually unable to determine whihc one is correct or not... and in the worst case, it might even lead to long resolving/lookup times as it needs to ask every nameserver about the existence of the record.

Reply:
libc restrictions...